libcrux-ml-kem/c/libcrux_mlkem_avx2.c

```diff
@@ -1650,6 +1526,47 @@ static KRML_MUSTINLINE void shake128_squeeze_three_blocks_a9_4d1(
   shake128_squeeze_three_blocks_6b1(self, ret);
 }
 
+/**
+ If `bytes` contains a set of uniformly random bytes, this function
+ uniformly samples a ring element `â` that is treated as being the NTT
+ representation of the corresponding polynomial `a`.
+
+ Since rejection sampling is used, it is possible the supplied bytes are
+ not enough to sample the element, in which case an `Err` is returned and the
+ caller must try again with a fresh set of bytes.
+
+ This function <strong>partially</strong> implements <strong>Algorithm
+ 6</strong> of the NIST FIPS 203 standard, We say "partially" because this
+ implementation only accepts a finite set of bytes as input and returns an error
+ if the set is not enough; Algorithm 6 of the FIPS 203 standard on the other
+ hand samples from an infinite stream of bytes until the ring element is filled.
+ Algorithm 6 is reproduced below:
+
+ ```plaintext
+ Input: byte stream B ∈ 𝔹*.
+ Output: array â ∈ ℤ₂₅₆.
```

Comment:
I hope C tools are ok with all the unicode characters 😬

Reply:
so far so good!